### PR TITLE
Ensure Microsoft.AspNetCore.Razor.RuntimeCompilation has the right build targets

### PR DESCRIPTION
**Pull request by pranavkm**
* Include build and build transitive folders in RuntimeCompilation package
* Use a different folder name to allow GitHub indexing. See https://github.com/aspnet/AspNetCore/issues/14846
* Add a test to verify transitive build targets work

Fixes https://github.com/aspnet/AspNetCore/issues/14813
Fixes https://github.com/aspnet/AspNetCore/issues/12768

### CODE REVIEW

File: src/Mvc/Mvc.Razor.RuntimeCompilation/src/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.csproj [hunk 1]
```diff
@@ -23,6 +23,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <None Include="build\$(DefaultNetCoreTargetFramework)\*" Pack="true" PackagePath="build\$(DefaultNetCoreTargetFramework)" />
+    <None Include="targets\$(PackageId).targets" Pack="true" PackagePath="build\$(DefaultNetCoreTargetFramework)\$(PackageId).targets" />
```
Comment (ajaybhargavb) [inline src/Mvc/Mvc.Razor.RuntimeCompilation/src/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.csproj]: Who sets packageid? Also, are we renaming the folder name to netcoreapp3.1?

Comment (pranavkm) [reply src/Mvc/Mvc.Razor.RuntimeCompilation/src/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.csproj]: PackageId is set by the SDK. So this should result in

`<None Include="targets\Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.targets" Pack="true" PackagePath="build\netcoreapp3.1\Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.targets" />`

File: src/Mvc/test/WebSites/RazorBuildWebSite/RazorBuildWebSite.csproj [hunk 2]
```diff
@@ -21,4 +20,6 @@
     <Reference Include="Microsoft.AspNetCore.Diagnostics" />
   </ItemGroup>
 
+  <Import Project="..\..\..\Mvc.Razor.RuntimeCompilation\src\targets\Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.targets" />
```
Comment (ajaybhargavb) [inline src/Mvc/test/WebSites/RazorBuildWebSite/RazorBuildWebSite.csproj]: No `$(SolutionRoot)`?

**Review by ajaybhargavb — APPROVED** [review]

File: src/Mvc/test/WebSites/RazorBuildWebSite/RazorBuildWebSite.csproj (hunk 2 above)
Comment (pranavkm) [reply src/Mvc/test/WebSites/RazorBuildWebSite/RazorBuildWebSite.csproj]: Ehh, unless something like helix requires it

**Pull request merged**